python/tests/test_integration_schema.py

```diff
@@ -0,0 +1,2617 @@
+    (StringType(), BinaryType()),
+    (BinaryType(), StringType()),
+    (IntegerType(), LongType()),
+    (FloatType(), DoubleType()),
+    (DecimalType(9, 2), DecimalType(18, 2)),
+]
+
+
+@pytest.mark.parametrize("from_type, to_type", allowed_promotions, ids=str)
+@pytest.mark.integration
+def test_allowed_updates(from_type: PrimitiveType, to_type: PrimitiveType, catalog: Catalog) -> None:
+    tbl = _create_table_with_schema(
+        catalog,
+        Schema(
+            NestedField(field_id=1, name="bar", field_type=from_type, required=True),
+        ),
+    )
+
+    with tbl.update_schema() as schema_update:
+        schema_update.update_column("bar", to_type)
+
+    assert tbl.schema() == Schema(NestedField(field_id=1, name="bar", field_type=to_type, required=True))
+
+
+disallowed_promotions_types = [
+    BooleanType(),
+    IntegerType(),
+    LongType(),
+    FloatType(),
+    DoubleType(),
+    DateType(),
+    TimeType(),
+    TimestampType(),
+    TimestamptzType(),
+    StringType(),
+    UUIDType(),
+    BinaryType(),
+    FixedType(3),
+    FixedType(4),
+    # We'll just allow Decimal promotions right now
+    # https://github.com/apache/iceberg/issues/8389
+    # DecimalType(9, 2),
+    # DecimalType(9, 3),
+    # DecimalType(18, 2)
+]
+
+
+@pytest.mark.parametrize("from_type", disallowed_promotions_types, ids=str)
+@pytest.mark.parametrize("to_type", disallowed_promotions_types, ids=str)
+@pytest.mark.integration
+def test_disallowed_updates(from_type: PrimitiveType, to_type: PrimitiveType, catalog: Catalog) -> None:
+    tbl = _create_table_with_schema(
+        catalog,
+        Schema(
+            NestedField(field_id=1, name="bar", field_type=from_type, required=True),
+        ),
+    )
+
+    if from_type != to_type and (from_type, to_type) not in allowed_promotions:
+        with pytest.raises(ValidationError) as exc_info:
+            with tbl.update_schema() as schema_update:
+                schema_update.update_column("bar", to_type)
+
+        assert str(exc_info.value).startswith("Cannot change column type: bar:")
+    else:
+        with tbl.update_schema() as schema_update:
+            schema_update.update_column("bar", to_type)
+
+        assert tbl.schema() == Schema(
+            NestedField(field_id=1, name="bar", field_type=to_type, required=True),
+        )
+
+
+@pytest.mark.integration
+def test_rename_simple(simple_table: Table) -> None:
+    with simple_table.update_schema() as schema_update:
+        schema_update.rename_column("foo", "vo")
+
+    assert simple_table.schema() == Schema(
+        NestedField(field_id=1, name="vo", field_type=StringType(), required=False),
+        NestedField(field_id=2, name="bar", field_type=IntegerType(), required=True),
+        NestedField(field_id=3, name="baz", field_type=BooleanType(), required=False),
+        schema_id=1,
+        identifier_field_ids=[2],
+    )
+
+
+@pytest.mark.integration
+def test_rename_simple_nested(catalog: Catalog) -> None:
```

Comment:
Would be nice to have a case with `.` names, like `rename_column(("a.b", "c.d"), "e.f")` produces column `("a.b", "e.f")`.

Reply:
Added